internal/beater/processors.go

```diff
@@ -74,3 +80,52 @@ func newObserverBatchProcessor() model.ProcessBatchFunc {
 		return nil
 	}
 }
+
+func newDocappenderBatchProcessor(a *docappender.Appender) model.ProcessBatchFunc {
+	var pool sync.Pool
+	pool.New = func() any {
+		return &pooledReader{pool: &pool}
+	}
+	return func(ctx context.Context, b *model.Batch) error {
+		for _, event := range *b {
+			r := pool.Get().(*pooledReader)
+			if err := event.MarshalFastJSON(&r.jsonw); err != nil {
+				r.reset()
+				return err
+			}
+			r.indexBuilder.WriteString(event.DataStream.Type)
+			r.indexBuilder.WriteByte('-')
+			r.indexBuilder.WriteString(event.DataStream.Dataset)
+			r.indexBuilder.WriteByte('-')
+			r.indexBuilder.WriteString(event.DataStream.Namespace)
+			index := r.indexBuilder.String()
+			if err := a.Add(ctx, index, r); err != nil {
+				r.reset()
```

Comment:
The reader is not calling `reset` on success and thus not releasing `indexBuilder` and `jsonw` back to the pool. Is that expected?

Reply:
Yes, in the success case the reader is in the docappender's queue. When the document body is consumed, the reader's WriteTo method will be called, and that will call reset.